app/tests/factories.py

```diff
@@ -0,0 +1,197 @@
+'''
+NoI test factories
+
+Factories to create test objects
+'''
+
+from app import QUESTIONS_BY_ID, ORG_TYPES, LOCALES
+from app.models import (db, User, UserSkill, UserJoinedEvent,
+                        UserExpertiseDomain, UserLanguage, Email,
+                        ConnectionEvent, SharedMessageEvent)
+
+from flask import current_app
+
+from factory import LazyAttribute, RelatedFactory, post_generation
+from factory.alchemy import SQLAlchemyModelFactory
+from factory.fuzzy import FuzzyChoice, FuzzyNaiveDateTime
+from faker import Faker
+from datetime import datetime
+from random import choice, randint, sample
+
+import logging
+
+
+# Only log warnings from factoryboy, its default DEBUG statements are spammy
+logger = logging.getLogger('factory')
+logger.setLevel(logging.WARNING)
+fake = Faker()
+
+class SharedMessageEventFactory(SQLAlchemyModelFactory):
+
+    class Meta:  # pylint: disable=old-style-class,no-init,too-few-public-methods
+        model = SharedMessageEvent
+        sqlalchemy_session = db.session
+
+    message = LazyAttribute(lambda o: fake.paragraph())
+
+
+class EmailFactory(SQLAlchemyModelFactory):
+
+    class Meta:  # pylint: disable=old-style-class,no-init,too-few-public-methods
+        model = Email
+        sqlalchemy_session = db.session
+
+
+class ConnectionEventFactory(SQLAlchemyModelFactory):
+
+    class Meta:  # pylint: disable=old-style-class,no-init,too-few-public-methods
+        model = ConnectionEvent
+        sqlalchemy_session = db.session
+
+
+class UserJoinedEventFactory(SQLAlchemyModelFactory):
+
+    class Meta:  # pylint: disable=old-style-class,no-init,too-few-public-methods
+        model = UserJoinedEvent
+        sqlalchemy_session = db.session
+
+
+class UserExpertiseDomainFactory(SQLAlchemyModelFactory):
+
+    class Meta:  # pylint: disable=old-style-class,no-init,too-few-public-methods
+        model = UserExpertiseDomain
+        sqlalchemy_session = db.session
+
+    name = LazyAttribute(lambda o: choice(current_app.config['DOMAINS']))
+
+
+class UserLanguageFactory(SQLAlchemyModelFactory):
+
+    class Meta:  # pylint: disable=old-style-class,no-init,too-few-public-methods
+        model = UserLanguage
+        sqlalchemy_session = db.session
+
+    locale = FuzzyChoice(LOCALES)
+
+
+class UserSkillFactory(SQLAlchemyModelFactory): # pylint: disable=no-init,too-few-public-methods
+
+    class Meta:  # pylint: disable=old-style-class,no-init,too-few-public-methods
+        model = UserSkill
+        sqlalchemy_session = db.session
+
+    name = FuzzyChoice(QUESTIONS_BY_ID.keys())
+    #level = FuzzyChoice([l['score'] for l in LEVELS.values()])
```

Comment:
Yep, it was throwing a strange error.  You can try reproducing yourself by swapping for the next line.